src/Cli/dotnet/commands/dotnet-workload/install/WorkloadManifestUpdater.cs

```diff
@@ -291,40 +293,33 @@ private IEnumerable<ManifestId> GetInstalledManifestIds()
             return _workloadResolver.GetInstalledManifests().Select(manifest => new ManifestId(manifest.Id));
         }
 
-        private async Task UpdateAdvertisingManifestAsync(ManifestId manifestId, bool includePreviews, DirectoryPath? offlineCache = null)
+        private async Task UpdateAdvertisingManifestAsync(WorkloadManifestInfo manifest, bool includePreviews, DirectoryPath? offlineCache = null)
         {
             string packagePath = null;
             string extractionPath = null;
+            var manifestId = new ManifestId(manifest.Id);
+            string currentFeatureBand = _sdkFeatureBand.ToString();
 
             try
             {
                 var adManifestPath = GetAdvertisingManifestPath(_sdkFeatureBand, manifestId);
-                if (offlineCache == null || !offlineCache.HasValue)
+                
+                bool success;
+                (success, packagePath) = await GetManifestPackageUpdate(_sdkFeatureBand, manifestId, includePreviews, offlineCache);
+                if (!success)
                 {
-                    try
-                    {
-                        packagePath = await _nugetPackageDownloader.DownloadPackageAsync(
-                            GetManifestPackageId(_sdkFeatureBand, manifestId),
-                            packageSourceLocation: _packageSourceLocation,
-                            includePreview: includePreviews);
-                    }
-                    catch (NuGetPackageNotFoundException)
+                    if (!(manifest.ManifestFeatureBand).Equals(_sdkFeatureBand))
                     {
-                        _reporter.WriteLine(string.Format(LocalizableStrings.AdManifestPackageDoesNotExist, manifestId));
+                        (success, packagePath) = await GetManifestPackageUpdate(new SdkFeatureBand(manifest.ManifestFeatureBand), manifestId, includePreviews, offlineCache);
+                        currentFeatureBand = manifest.ManifestFeatureBand.ToString();
                     }
                 }
-                else
+                if (!success)
```

Comment:
Is the second `if` needed? It appears to be inside the same block as the `if (!success)` on line 309